src/tsickle_compiler_host.ts

```diff
@@ -0,0 +1,199 @@
+      case Pass.Tsickle:
+        return this.runTsickle(sourceFile, this.oldProgram, fileName, languageVersion);
+      default:
+        throw new Error('tried to use TsickleCompilerHost with unknown pass enum');
+    }
+  }
+
+  writeFile(
+      fileName: string, content: string, writeByteOrderMark: boolean,
+      onError?: (message: string) => void, sourceFiles?: ts.SourceFile[]): void {
+    fileName = this.delegate.getCanonicalFileName(fileName);
+    if (this.options.googmodule && !fileName.match(/\.d\.ts$/)) {
+      content = this.convertCommonJsToGoogModule(fileName, content);
+    }
+
+    this.delegate.writeFile(fileName, content, writeByteOrderMark, onError, sourceFiles);
+  }
+
+  convertCommonJsToGoogModule(fileName: string, content: string): string {
+    // Strip off the file name extension.
+    fileName = fileName.substring(0, fileName.lastIndexOf('.'));
+
+    let {output, referencedModules} = processES5(
+        fileName, fileName, content, this.environment.pathToModuleName.bind(this.environment),
+        this.options.es5Mode, this.options.prelude);
+
+    const moduleName = this.environment.pathToModuleName('', fileName);
+    this.modulesManifest.addModule(fileName, moduleName);
+    for (let referenced of referencedModules) {
+      this.modulesManifest.addReferencedModule(fileName, referenced);
+    }
+
+    return output;
+  }
+
+  private runDecoratorDownlevel(
+      sourceFile: ts.SourceFile, program: ts.Program, fileName: string,
+      languageVersion: ts.ScriptTarget): ts.SourceFile {
+    if (this.environment.shouldSkipTsickleProcessing(fileName)) return sourceFile;
+    let fileContent = sourceFile.text;
+    const converted = convertDecorators(program.getTypeChecker(), sourceFile);
+    if (converted.diagnostics) {
+      this.diagnostics.push(...converted.diagnostics);
+    }
+    if (converted.output === fileContent) {
+      // No changes; reuse the existing parse.
+      return sourceFile;
+    }
+    fileContent = converted.output + this.ANNOTATION_SUPPORT;
+    return ts.createSourceFile(fileName, fileContent, languageVersion, true);
+  }
+
+  private runTsickle(
+      sourceFile: ts.SourceFile, program: ts.Program, fileName: string,
+      languageVersion: ts.ScriptTarget): ts.SourceFile {
+    let isDefinitions = /\.d\.ts$/.test(fileName);
+    // Don't tsickle-process any d.ts that isn't a compilation target;
+    // this means we don't process e.g. lib.d.ts.
+    if (isDefinitions && this.environment.shouldSkipTsickleProcessing(fileName)) return sourceFile;
+
+    let {output, externs, diagnostics} =
+        annotate(program, sourceFile, {untyped: !this.options.tsickleTyped});
+    if (externs) {
+      this.externs[fileName] = externs;
+    }
+    if (this.environment.shouldIgnoreWarningsForPath(sourceFile.path)) {
+      // All diagnostics (including warnings) are treated as errors.
+      // But for third-party code we won't fix warnings so just discard them.
```

Comment:
I agree it's generic; my objection is that we're talking about third-party code here when the function we're using doesn't say anything about third-party.  (If anything, the mention of third part should be on the definition of the interface as an example of the sorts of code you might want to ignore warnings on)